demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+                if (!FLAGS_no_show && FLAGS_delay != -1) {
+                    cv::imshow("main", showMat);
+                    key = static_cast<char>(cv::waitKey(delay));
+                }
+
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    latencySum += ((processingEndTimes[i] - completedIRInfo.images[i].startTime)
+                                   / cv::getTickFrequency());
+                }
+                avgLatency = latencySum / framesNum;
+                accuracy = static_cast<double>(rightPredictionsCount) / totalPredictionsCount;
+                gridMat.textUpdate(avgFPS, avgLatency, accuracy, isTestMode, hasClasses, presenter);
+            }
+            else if (!emptyInferRequests.empty()) {
+                mutex.unlock();
+                cv::Mat nextImage = inputImages[nextImageIndex];
+                long long IRstartTime = cv::getTickCount();
+                resizeImage(nextImage, modelInputResolution);
+                emptyInferRequests.front().images.push_back(
+                    IRInfo::IRImage(nextImage,
+                                    std::stoul(hasClasses ? classIndexes[nextImageIndex] : "0"),
+                                    IRstartTime));
+                nextImageIndex++;
+                if (nextImageIndex == inputImagesCount) {
+                    nextImageIndex = 0;
+                }
+                if (emptyInferRequests.front().images.size() == FLAGS_b) {
+                    auto emptyIR = emptyInferRequests.front();
+                    auto& ir = emptyIR.ir;
+                    auto irImages = emptyIR.images;
+
+                    emptyIR.ir.SetCompletionCallback([&ir,
+                                                      irImages,
+                                                      &completedInferRequests,
+                                                      &mutex,
+                                                      &condVar,
+                                                      &irCallbackException] {
+                        try {
+                            {
+                                std::lock_guard<std::mutex> lock(mutex);
+
+                                completedInferRequests.push(IRInfo(ir, irImages));
+                            }
+                            condVar.notify_one();
+                        }
+                        catch(...) {
+                            irCallbackException = std::current_exception();
+                        }
+                    });
+                    
+                    auto inputBlob = emptyIR.ir.GetBlob(inputBlobName);
+                    for (unsigned i = 0; i < FLAGS_b; i++) {
+                        matU8ToBlob<uint8_t>(emptyIR.images[i].mat, inputBlob, i);
+                    }
+                    emptyIR.ir.StartAsync();
+                    emptyInferRequests.pop();
+                }
+            }
+            
+            mutex.lock();
```

Comment:
You should lock `lock` instead of `mutex` because that's what you pass to `wait()`.